idea/src/org/jetbrains/kotlin/idea/quickfix/expectactual/AbstractCreateDeclarationFix.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2010-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.idea.quickfix.expectactual
+
+import com.intellij.ide.util.EditorHelper
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.module.Module
+import com.intellij.openapi.project.DumbService
+import com.intellij.openapi.project.Project
+import com.intellij.psi.codeStyle.CodeStyleManager
+import org.jetbrains.kotlin.idea.core.ShortenReferences
+import org.jetbrains.kotlin.idea.quickfix.KotlinQuickFixAction
+import org.jetbrains.kotlin.idea.refactoring.introduce.showErrorHint
+import org.jetbrains.kotlin.idea.util.application.executeWriteCommand
+import org.jetbrains.kotlin.psi.*
+
+abstract class AbstractCreateDeclarationFix<D : KtNamedDeclaration>(
+    declaration: D,
+    protected val module: Module,
+    protected val generateIt: KtPsiFactory.(Project, D) -> D?
+) : KotlinQuickFixAction<D>(declaration) {
+
+    override fun getFamilyName(): String = "Create expected / actual declaration"
+
+    protected val elementType: String = element.getTypeDescription()
+
+    override fun startInWriteAction() = false
+
+    protected abstract fun KtFile.getFileForGeneratingDeclaration(originalDeclaration: KtNamedDeclaration): KtFile?
```

Comment:
Probably broken English, `ForGeneratedDeclaration`?

Reply:
I mean here "get file to generate declaration inside"

Reply:
May be it's wrong, will recheck

Reply:
I gave function different name (`findExistingFileToCreateDeclaration`) to avoid misunderstanding